plugins/tracing/client.go

```diff
@@ -0,0 +1,190 @@
+/*
+Copyright 2017 Caicloud Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tracing
+
+import (
+	"net/http"
+	"net/http/httptrace"
+
+	opentracing "github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+	tlog "github.com/opentracing/opentracing-go/log"
+)
+
+// Transport wraps a RoundTripper. If a request is being traced with
+// Tracer, Transport will inject the current span into the headers,
+// and set HTTP related tags on the span.
+type Transport struct {
+	// The actual RoundTripper to use for the request. A nil
+	// RoundTripper defaults to http.DefaultTransport.
+	http.RoundTripper
+	// Enable the ClientTrace, See https://blog.golang.org/http-tracing for more.
+	EnableHTTPtrace bool
+	// The default is HTTP method
+	OperationName string
+}
+
+// RoundTrip implements the RoundTripper interface.
+func (t *Transport) RoundTrip(req *http.Request) (*http.Response, error) {
+	rt := t.RoundTripper
+	if rt == nil {
+		rt = http.DefaultTransport
+	}
+
+	operationName := t.OperationName
+	if operationName == "" {
+		operationName = req.Method
+	}
+
+	span, ctx := StartSpanFromContext(req.Context(), "HTTP: "+operationName)
```

Comment:
my understanding is that roundtrip  will start a new span if there is no span found in the context?

Reply:
discussed offline, we don't use global tracer, so it's not necessary